packages/toast/src/ControlledToast/ControlledToast.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from 'react';
+import defaultsDeep from 'lodash/defaultsDeep';
+
+import { defaultToastProps } from '../InternalToast/defaultProps';
+import { ToastId, useToast } from '../ToastContext';
+
+import { ControlledToastProps } from './ControlledToast.types';
+
+/**
+ * A controlled toast component.
+ *
+ * To use this component, you must manage the `open` state yourself.
+ *
+ * It's recommended to use the hook `useToast` instead
+ */
+export const ControlledToast = ({ open, ...props }: ControlledToastProps) => {
+  props = defaultsDeep(props, defaultToastProps);
+  const { pushToast, popToast } = useToast();
+  const [toastId, setToastId] = useState<ToastId | null>(null);
+
+  useEffect(() => {
+    if (open && !toastId) {
+      const _id = pushToast({ isControlled: true, ...props });
```

Comment:
nit: why `_id`

Reply:
just to make it clear it's a temp variable